pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -774,7 +774,11 @@ brokerId, getLogEventTag(data), serviceUnit,
         }
     }
 
-    private void handleSkippedEvent(String serviceUnit) {
+    private void handleSkippedEvent(String serviceUnit, ServiceUnitStateData skippedData) {
+        if (skippedData.state() == Free) {
```

Comment:
This is risky as the skipped message impacts the current states.

The channel design principle is that only non-skipped messages should impact the states in this channel.